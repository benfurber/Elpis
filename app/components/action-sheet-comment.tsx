import React, { Component } from "react";
import ActionSheet from "react-native-actionsheet";

import { Comment, NavigationType } from "interfaces";
import { labels } from "labels";

interface Props {
  item: Comment;
  navigation: NavigationType;
  refProp: any;
}

class ActionSheetComment extends Component<Props> {
  onPress(index) {
    if (index === 1) console.log("Delete");
    if (index === 2) console.log("Update");
  }

  render() {
    const { refProp } = this.props;
    const { cancel } = labels;

    const options = [cancel, "Delete comment", "Edit comment"];

    return (
      <ActionSheet
        ref={refProp}
        options={options}
        cancelButtonIndex={0}
        onPress={index => this.onPress(index)}
      />
    );
  }
}

export { ActionSheetComment };
