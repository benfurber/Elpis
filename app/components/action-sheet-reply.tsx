import React, { Component } from "react";
import ActionSheet from "react-native-actionsheet";

import { NavigationType, Reply } from "interfaces";
import { labels } from "labels";
import { DELETE_REPLY } from "mutations";
import { client } from "utils";

interface Props {
  item: Reply;
  navigation: NavigationType;
  refProp: any;
}

class ActionSheetReply extends Component<Props> {
  async actionSheetOnPress(index) {
    if (index === 1) {
      const { id } = this.props.item;
      this.setState({ deleting: true });

      await client.mutate({
        mutation: DELETE_REPLY,
        variables: {
          id,
        },
      });
    }

    if (index === 2) {
      const { item, navigation } = this.props;
      const { comment, content, id, imagePath } = item;

      const commentId = comment.id;
      const currentReply = { content, id, imagePath };

      navigation.navigate("AddReply", { commentId, currentReply });
    }
  }

  render() {
    const { refProp } = this.props;
    const { cancel, deleteYourReply, editReply } = labels;

    return (
      <ActionSheet
        ref={refProp}
        options={[cancel, deleteYourReply, editReply]}
        cancelButtonIndex={0}
        onPress={index => this.actionSheetOnPress(index)}
      />
    );
  }
}

export { ActionSheetReply };
