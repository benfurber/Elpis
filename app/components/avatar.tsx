import React, { Component } from "react";
import { Image, View } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";

import { Author, NavigationType } from "interfaces";
import { elements } from "styles";

const annonProfilePath = "../assets/images/profile-pic-annon.png";

interface Props {
  avatarPath: null | Author["avatarPath"];
  containerStyles?: object;
  navigation?: NavigationType;
  userId?: Author["id"];
  size?: "small" | "medium" | "large" | "xl" | "feature";
  styles?: object;
}

class Avatar extends Component<Props> {
  onPress() {
    const { navigation, userId } = this.props;

    if (navigation && userId) {
      navigation.navigate("UserProfile", {
        navigation,
        userId,
      });
    }
  }

  render() {
    const { avatarPath, containerStyles, size, styles } = this.props;

    const path = avatarPath ? { uri: avatarPath } : require(annonProfilePath);

    const styleSet = {
      feature: elements.imageRoundFeature,
      large: elements.imageRoundLarge,
      medium: elements.imageRound,
      small: elements.imageRoundSmall,
      xl: elements.imageRoundXL,
    };

    return (
      <View style={containerStyles}>
        <TouchableOpacity onPress={() => this.onPress()}>
          <Image source={path} style={[styleSet[size || "medium"], styles]} />
        </TouchableOpacity>
      </View>
    );
  }
}

export { Avatar };
