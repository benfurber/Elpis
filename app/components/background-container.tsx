import React from "react";
import { ImageBackground, SafeAreaView, StyleSheet, View } from "react-native";

import { colours } from "styles";

interface Props {
  children: object;
  style?: object;
  viewStyle?: object;
}

const BackgroundContainer = (props: Props) => {
  return (
    <ImageBackground
      source={require("../assets/images/background.png")}
      style={props.style || styles.container}
    >
      <SafeAreaView style={styles.safeArea}>
        <View style={props.viewStyle || styles.viewContainer}>
          {props.children}
        </View>
      </SafeAreaView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colours.transparentBlue,
    height: "100%",
    width: "100%",
  },
  safeArea: {
    flex: 1,
  },
  viewContainer: {
    flex: 1,
    flexDirection: "column",
  },
});

export { BackgroundContainer };
