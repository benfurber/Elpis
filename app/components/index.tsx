export { Comments } from "./comments";
export { BackgroundContainer } from "./background-container";
export { BackgroundModal } from "./background-modal";
export { Badge } from "./badge";
export { ButtonSubmit } from "./button-submit";
export { FormAddProfilePicture } from "./form-add-profile-picture";
export { FormAddPassword } from "./form-add-password";
export { Icon } from "./icon";
export { LoginForm } from "./login-form";
export { Logo } from "./logo";
export { MessageBox } from "./message-box";
export { Post } from "./post";
export { ProfilePictureField } from "./profile-picture-field";
export { TextField } from "./text-field";
export { TextInput } from "./text-input";
export { Title } from "./title";
