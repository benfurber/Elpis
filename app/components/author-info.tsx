import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";

import { Avatar, Title } from "components";
import { Comment, NavigationType } from "interfaces";
import { colours, elements, layout } from "styles";
import { formatDate } from "utils";

interface Props {
  item: Comment;
  navigation: NavigationType;
}

class AuthorInfo extends Component<Props> {
  render() {
    const { item, navigation } = this.props;
    const { author, publishedAt, title } = item;

    return (
      <View style={styles.container}>
        <View style={styles.avatarContainer}>
          <Avatar
            avatarPath={author.avatarPath}
            navigation={navigation}
            userId={author.id}
            size="large"
          />
        </View>
        <View style={styles.titleContainer}>
          {title && <Title text={title} />}
          <Text style={styles.metaDetails}>
            <Text style={styles.name}>{`${author.name} `}</Text>
            {formatDate(publishedAt, false)}
          </Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  avatarContainer: {
    alignItems: "center",
    flexDirection: "row",
    width: 70,
  },
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  metaDetails: {
    marginBottom: layout.spacing,
    ...elements.textDate,
  },
  name: {
    color: colours.navyBlueDark,
    fontWeight: "bold",
  },
  text: {
    flex: 1,
    flexWrap: "nowrap",
    marginRight: layout.spacingXL,
  },
  titleContainer: {
    flex: 1,
  },
});

export { AuthorInfo };
