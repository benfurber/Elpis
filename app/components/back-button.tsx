import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import { Icon } from "components";
import { NavigationType } from "interfaces";
import { labels } from "labels";
import { layout } from "styles";

interface Props {
  navigation: NavigationType;
  onPress?: () => void;
  text?: string;
}

const BackButton = (props: Props) => {
  const { text, navigation } = props;
  const { back } = labels.back;

  const onPress = () => navigation.pop();

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress || onPress}
    >
      <Icon name="angle-double-left" style={styles.icon} />
      <Text>{text || back}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    padding: layout.spacing,
  },
  icon: {
    paddingRight: layout.spacingS,
  },
});

export { BackButton };
