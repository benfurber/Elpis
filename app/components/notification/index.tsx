import React, { Component } from "react";
import { StyleSheet, TouchableOpacity, View } from "react-native";

import { NavigationType, Notification as NotificationType } from "interfaces";
import { colours, layout } from "styles";

import { Avatar } from "./avatar";
import { Details } from "./details";
import { Title } from "./title";

interface Props {
  item: NotificationType;
  navigation: NavigationType;
}

class Notification extends Component<Props> {
  onPress = () => {
    const { content, type, postId } = this.props.item;
    const { id } = content;

    const postParams = {
      id,
      setDisplay: "body",
    };
    const replyParams = {
      id: postId,
      commentId: id,
      setDisplay: "comments",
    };

    const params = type === "comment" ? replyParams : postParams;
    return this.props.navigation.navigate("Post", params);
  };

  render() {
    return (
      <TouchableOpacity style={styles.item} onPress={this.onPress}>
        <View style={styles.row}>
          <Avatar item={this.props.item} />
          <Title item={this.props.item} />
        </View>
        <Details item={this.props.item} />
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  item: {
    backgroundColor: colours.whiteTransparent,
    marginBottom: layout.spacing,
    padding: layout.spacing,
  },
  row: {
    alignItems: "center",
    flexDirection: "row",
  },
});

export { Notification };
