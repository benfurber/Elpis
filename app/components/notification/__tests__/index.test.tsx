import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { mockDateNow } from "../../../test-utils";

import { Notification } from "components";

let navigation;
jest.mock(navigation, () => jest.fn());

describe("Notification", () => {
  describe("new", () => {
    it("for comment type it renders correctly", () => {
      mockDateNow("2019-10-09");

      const commentNotification = {
        content: {
          author: {
            name: "May",
            avatarPath: "",
          },
          content:
            "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
          createdAt: new Date("2019-09-23"),
          id: "93716",
        },
        newNotification: true,
        type: "comment",
        postId: "b845q3",
      };

      const component = shallow(
        <Notification item={commentNotification} navigation={navigation} />,
      );

      expect(component).toMatchSnapshot();
    });

    it("for post type it renders correctly", () => {
      mockDateNow("2019-10-09");

      const postNotification = {
        content: {
          author: {
            name: "E2M",
            avatarPath: "",
          },
          content:
            "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
          createdAt: new Date("2019-09-23"),
          id: "986sasdfd",
          imagePath: "",
        },
        newNotification: true,
        type: "post",
      };

      const component = shallow(
        <Notification item={postNotification} navigation={navigation} />,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe("old", () => {
    it("for comment type it renders correctly", () => {
      mockDateNow("2019-10-09");

      const commentNotification = {
        content: {
          author: {
            name: "May",
            avatarPath: "",
          },
          content:
            "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
          createdAt: new Date("2019-09-23"),
          id: "315928bjaf",
        },
        newNotification: false,
        type: "comment",
        postId: "35245gt1",
      };

      const component = shallow(
        <Notification item={commentNotification} navigation={navigation} />,
      );

      expect(component).toMatchSnapshot();
    });

    it("for post type it renders correctly", () => {
      mockDateNow("2019-10-09");

      const postNotification = {
        content: {
          author: {
            name: "E2M",
            avatarPath: "",
          },
          content:
            "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
          createdAt: new Date("2019-09-23"),
          id: "078vw213",
          imagePath: "",
        },
        newNotification: false,
        type: "post",
      };

      const component = shallow(
        <Notification item={postNotification} navigation={navigation} />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
