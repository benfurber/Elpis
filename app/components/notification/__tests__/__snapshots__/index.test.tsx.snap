// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification new for comment type it renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
      "marginBottom": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <NotificationAvatar
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "May",
            },
            "content": "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
            "createdAt": 2019-09-23T02:24:00.000Z,
            "id": "93716",
          },
          "newNotification": true,
          "type": "comment",
        }
      }
    />
    <NotificationTitle
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "May",
            },
            "content": "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
            "createdAt": 2019-09-23T02:24:00.000Z,
            "id": "93716",
          },
          "newNotification": true,
          "type": "comment",
        }
      }
    />
  </View>
  <Details
    item={
      Object {
        "content": Object {
          "author": Object {
            "avatarPath": "",
            "name": "May",
          },
          "content": "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
          "createdAt": 2019-09-23T02:24:00.000Z,
          "id": "93716",
        },
        "newNotification": true,
        "type": "comment",
      }
    }
  />
</TouchableOpacity>
`;

exports[`Notification new for post type it renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
      "marginBottom": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <NotificationAvatar
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "E2M",
            },
            "content": "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
            "createdAt": 2019-09-23T00:00:00.000Z,
            "id": "986sasdfd",
            "imagePath": "",
          },
          "newNotification": true,
          "type": "post",
        }
      }
    />
    <NotificationTitle
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "E2M",
            },
            "content": "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
            "createdAt": 2019-09-23T00:00:00.000Z,
            "id": "986sasdfd",
            "imagePath": "",
          },
          "newNotification": true,
          "type": "post",
        }
      }
    />
  </View>
  <Details
    item={
      Object {
        "content": Object {
          "author": Object {
            "avatarPath": "",
            "name": "E2M",
          },
          "content": "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
          "createdAt": 2019-09-23T00:00:00.000Z,
          "id": "986sasdfd",
          "imagePath": "",
        },
        "newNotification": true,
        "type": "post",
      }
    }
  />
</TouchableOpacity>
`;

exports[`Notification old for comment type it renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
      "marginBottom": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <NotificationAvatar
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "May",
            },
            "content": "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
            "createdAt": 2019-09-23T02:24:00.000Z,
            "id": "315928bjaf",
          },
          "newNotification": false,
          "type": "comment",
        }
      }
    />
    <NotificationTitle
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "May",
            },
            "content": "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
            "createdAt": 2019-09-23T02:24:00.000Z,
            "id": "315928bjaf",
          },
          "newNotification": false,
          "type": "comment",
        }
      }
    />
  </View>
  <Details
    item={
      Object {
        "content": Object {
          "author": Object {
            "avatarPath": "",
            "name": "May",
          },
          "content": "Lucas ipsum dolor sit amet organa c-3p0 hutt anakin ponda biggs yoda moff jinn solo. Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Moff solo obi-wan antilles grievous lando mandalore.",
          "createdAt": 2019-09-23T02:24:00.000Z,
          "id": "315928bjaf",
        },
        "newNotification": false,
        "type": "comment",
      }
    }
  />
</TouchableOpacity>
`;

exports[`Notification old for post type it renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
      "marginBottom": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <NotificationAvatar
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "E2M",
            },
            "content": "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
            "createdAt": 2019-09-23T00:00:00.000Z,
            "id": "078vw213",
            "imagePath": "",
          },
          "newNotification": false,
          "type": "post",
        }
      }
    />
    <NotificationTitle
      item={
        Object {
          "content": Object {
            "author": Object {
              "avatarPath": "",
              "name": "E2M",
            },
            "content": "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
            "createdAt": 2019-09-23T00:00:00.000Z,
            "id": "078vw213",
            "imagePath": "",
          },
          "newNotification": false,
          "type": "post",
        }
      }
    />
  </View>
  <Details
    item={
      Object {
        "content": Object {
          "author": Object {
            "avatarPath": "",
            "name": "E2M",
          },
          "content": "Mara fisto yoda ben jade. Mace ben wedge jinn leia luke. Lando mandalore.",
          "createdAt": 2019-09-23T00:00:00.000Z,
          "id": "078vw213",
          "imagePath": "",
        },
        "newNotification": false,
        "type": "post",
      }
    }
  />
</TouchableOpacity>
`;
